motd/src/lib.rs

```diff
@@ -495,22 +512,18 @@ impl PamMotdResolutionStrategy {
             return Err(merr!("{:?} is not a directory", dir));
         }
 
-        for entry in fs::read_dir(&dir).map_err(|e| merr!("reading dir '{:?}': {:?}", dir, e))? {
-            let entry = entry.map_err(|e| merr!("getting dir entry for '{:?}': {:?}", dir, e))?;
-            let path = entry.path();
-
-            if path.is_symlink() {
-                continue;
-            }
-
-            if path.is_dir() && recursive {
-                if let Ok(res) = Self::find_file(recursive, path, fname) {
-                    return Ok(res);
-                }
-            } else if path.is_file() {
-                if path.file_name().map(|name| name == fname).unwrap_or(false) {
-                    return Ok(path);
-                }
+        let mut traversal = walkdir::WalkDir::new(&dir);
+        if !recursive {
```

Comment:
When would you ever want !recurisve?

Reply:
I was thinking it might be used to speed up the resolution since this traversal is pretty slow.